napari/components/_tests/test_layers_base.py

```diff
@@ -0,0 +1,20 @@
+import numpy as np
+import pytest
+from numpy import array
+
+from napari.layers.base import Layer
+
+
+@pytest.mark.parametrize(
+    'dims,nworld,nshape,expected',
+    [
+        ([2, 1, 0, 3], 4, 2, [0, 1]),
+        ([2, 1, 0, 3], 4, 3, [1, 0, 2]),
+        ([2, 1, 0, 3], 4, 4, [2, 1, 0, 3]),
+        ([0, 1, 2, 3, 4, 5, 6, 7], 4, 4, [0, 1, 2, 3, 4, 5, 6, 7]),
```

Comment:
Unsure about this one, I think there must be some constrainsts like `nword > len(dims)`,   I would need to better understand what those variables are